### PR TITLE
feat(nsis): add an option to disable compression

### PR DESCRIPTION
**Pull request by Legend-Master**

### CODE REVIEW

**Review by amrbashir — COMMENTED** [review]
I am not sure what this PR is trying to solve, it is more or less looks like a refactor to me.

**Comment by Legend-Master** [comment]
Before

image

After

image

**Comment by amrbashir** [comment]
That still not an issue, if user specifies null we default back to `lzma`, or maybe we need to actually make `null` disable compression

**Comment by Legend-Master** [comment]
Isn't this about correcting the schema?

**Comment by amrbashir** [comment]
You're right, but I think the schema is kinda correct, maybe just missing a documentation to explain that using `null` will default to `lzma`.

This PR has a breaking change because it changed the field from `Option<T>` to `T` and I am afraid some users may want to be able to disable compression and if that were to be requested, we will have to make another breaking change, so maybe we should take the opportunity to cover that case as well?

**Comment by Legend-Master** [comment]
Right, we can include no compression as an option here

File: .changes/nsis-no-compression.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"tauri-utils": patch::breaking
+"tauri-bundler": patch::breaking
+"tauri-cli": patch::breaking
+"@tauri-apps/cli": patch::breaking
+---
+
+Add an option to disable NSIS compression
```
Comment (amrbashir) [inline .changes/nsis-no-compression.md]:
```suggestion
---
"tauri-utils": patch::feat
"tauri-bundler": patch::feat
---

Add an option to disable NSIS compression `bundle > nsis > compression: "none"`
```

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thank you

**Pull request merged**